From 9d0a72ab9aebdfb89bc38259b6c7715be2ac6e4c Mon Sep 17 00:00:00 2001
From: Kurt Lee <kurt.lee@cypress.com>
Date: Wed, 26 Jan 2022 03:27:41 -0600
Subject: [PATCH 113/208] Revert "brcmfmac: Improve the delay during scan"

This reverts commit df4722113f8d542e6f72ed5bddff639a487a7ba5.
After reviewed code, this change may affect auto channel selection in AP
mode, so revert this commit first and searching for better solution.
---
 drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c
index 9bc23fbc3c92..63bdb5115237 100644
--- a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c
+++ b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c
@@ -8491,9 +8491,6 @@ brcmf_cfg80211_dump_survey(struct wiphy *wiphy, struct net_device *ndev,
 
 	brcmf_dbg(TRACE, "Enter: channel idx=%d\n", idx);
 
-	if (!brcmf_is_apmode(ifp->vif))
-		return -ENOENT;
-
 	/* Do not run survey when VIF in CONNECTING / CONNECTED states */
 	if ((test_bit(BRCMF_VIF_STATUS_CONNECTING, &ifp->vif->sme_state)) ||
 	    (test_bit(BRCMF_VIF_STATUS_CONNECTED, &ifp->vif->sme_state))) {
-- 
2.17.1

